types/simple-peer/index.d.ts

```diff
@@ -1,40 +1,77 @@
+// Type definitions for simple-peer 9.11
 // Project: https://github.com/feross/simple-peer
 // Definitions by: Tomasz Łaziuk <https://github.com/tlaziuk>
 //                 xWiiLLz <https://github.com/xWiiLLz>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.1
+// TypeScript Version: 3.0
 
 /// <reference types="node" />
 
-import * as stream from 'stream';
+import * as stream from "stream";
 
 declare const SimplePeer: SimplePeer.SimplePeer;
 
 declare namespace SimplePeer {
     interface Options {
-        initiator?: boolean; // set to true if this is the initiating peer
-        channelConfig?: {}; // custom webrtc data channel configuration (used by createDataChannel)
-        channelName?: string; // custom webrtc data channel name
-        config?: {}; // custom webrtc configuration (used by RTCPeerConnection constructor)
-        constraints?: {}; // custom webrtc video/voice constraints (used by RTCPeerConnection constructor)
-        offerConstraints?: {}; // custom offer constraints (used by createOffer method)
-        answerConstraints?: {}; // custom answer constraints (used by createAnswer method)
-        sdpTransform?<T extends any>(sdp: T): T; // function to transform the generated SDP signaling data (for advanced users)
-        stream?: MediaStream; // if video/voice is desired, pass stream returned from getUserMedia
-        streams?: MediaStream[]; // an array of MediaStreams returned from getUserMedia
-        trickle?: boolean; // set to false to disable trickle ICE and get a single 'signal' event (slower)
-        wrtc?: {}; // RTCPeerConnection/RTCSessionDescription/RTCIceCandidate
-        objectMode?: boolean; // set to true to create the stream in Object Mode. In this mode, incoming string data is not automatically converted to Buffer objects.
+        /** set to `true` if this is the initiating peer */
+        initiator?: boolean;
+        /** custom webrtc data channel configuration (used by [`createDataChannel`](https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/createDataChannel)) */
+        channelConfig?: RTCDataChannelInit;
+        /** custom webrtc data channel name */
+        channelName?: string;
+        /** custom webrtc configuration (used by [`RTCPeerConnection`](https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection) constructor) */
+        config?: RTCConfiguration;
+        /** custom offer options (used by [`createOffer`](https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/createOffer) method) */
+        offerOptions?: RTCOfferOptions;
+        /** custom answer options (used by [`createAnswer`](https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/createAnswer) method) */
+        answerOptions?: RTCAnswerOptions;
+        /** function to transform the generated SDP signaling data (for advanced users) */
+        sdpTransform?: (this: Instance, sdp: string) => string;
```

Comment:
sdp is a string, see https://developer.mozilla.org/en-US/docs/Web/API/RTCSessionDescription/sdp